tests/Unit/HttpRepository/AirtableHttpRepositoryTest.php

```diff
@@ -0,0 +1,309 @@
+        $this->assertRecordIdandCreatedTime($record);
+
+        /** @var Customer $customer */
+        $customer = $record->getFields();
+
+        $this->assertCustomer($customer);
+    }
+
+    /** @test */
+    public function findOneByIdWillReturnObjectIfDataClassIsSet(): void
+    {
+        // Setup the dummy HttpClient
+        $httpClient = $this->createHttpClientMock(
+            '/MOCK_TABLE/MOCK_ID',
+            $this->getPayload()
+        );
+
+        // Given we have an airtable client
+        $airtableHttpRepository = $this->getAirtableHttpRepository($httpClient);
+
+        // When we call findOneById with a data class
+        $record = $airtableHttpRepository->findOneById('MOCK_TABLE', 'MOCK_ID', Customer::class);
+
+        $this->assertRecordIdandCreatedTime($record);
+        // Then the result should be an AirtableRecord
+        static::assertInstanceOf(AirtableRecord::class, $record);
+        // And the fields should be instance of Customer
+        static::assertInstanceOf(Customer::class, $record->getFields());
+
+        /** @var Customer $customer */
+        $customer = $record->getFields();
+
+        $this->assertCustomer($customer);
+    }
+
+    /** @test */
+    public function addOneRecordWillReturnNull()
+    {
+        // Setup the dummy HttpClient
+        $httpClient = $this->createHttpClientMock(
+            '/MOCK_TABLE',
+            [],
+            'POST'
+        );
+
+        // Given we have an airtable client
+        $airtableHttpRepository = $this->getAirtableHttpRepository($httpClient);
+
+        // When we call addOneRecord with a data class
+        $record = $airtableHttpRepository->addOneRecord('MOCK_TABLE', [], Customer::class);
+
+        static::assertNull($record);
+    }
+
+    /** @test */
+    public function addOneRecordWillReturnObjectIfDataClassIsSet()
+    {
+        // Setup the dummy HttpClient
+        $httpClient = $this->createHttpClientMock(
+            '/MOCK_TABLE',
+            $this->getPayload(),
+            'POST'
+        );
+
+        // Given we have an airtable client
+        $airtableHttpRepository = $this->getAirtableHttpRepository($httpClient);
+
+        // When we call addOneRecord with a data class
+        $record = $airtableHttpRepository->addOneRecord('MOCK_TABLE', ['id' => 'MOCK_ID']);
+        $this->assertRecordIdandCreatedTime($record);
+        // Then the result should be an AirtableRecord
+        static::assertInstanceOf(AirtableRecord::class, $record);
+    }
+
+    private function assertCustomer(Customer $customer): void
+    {
+        static::assertEquals('MOCK_FIRST_NAME', $customer->firstName);
+        static::assertEquals('MOCK_LAST_NAME', $customer->lastName);
+        static::assertEquals(1, $customer->id);
+        static::assertEquals('1986-10-30', $customer->birthDay);
+    }
+
+    private function assertRecordIdandCreatedTime(AirtableRecord $record): void
```

Comment:
same here, we avoid duplication, having the same stuff in the same place make life easier :fire: